### PR TITLE
Add app filter

### PR DESCRIPTION
**Pull request by PanSpagetka**
Add app filter to the header. Since we don't have final mocks and interactions, this is more like alpha version.

#### TODO
delete app switcher

#### Screenshot

Screenshot from 2020-12-17 11-44-33

https://issues.redhat.com/browse/RHCLOUD-11234

### CODE REVIEW

File: src/js/App/Header/AppFilter.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import SAP from '../../../../static/images/platform-icons/SAP.svg';
+import insights from '../../../../static/images/platform-icons/Insights.svg';
+
+import './AppFilter.scss';
+
+const AppFilter = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [filterValue, setFilterValue] = useState('');
+
+  const appList = [
```
Comment (Hyperkid123) [inline src/js/App/Header/AppFilter.js:25]: We should get this list from the `main.yml` file and cache it like the left nav.

File: static/images/platform-icons/Insights.svg [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Hyperkid123) [inline static/images/platform-icons/Insights.svg:1]: These icons are available as static resources (`/apps/landing/fonts/Insights.svg`). Check the landing page for the image srouce.

File: src/js/App/Header/AppFilter.js (hunk 1 above)
Comment (karelhala) [reply src/js/App/Header/AppFilter.js:23]: The file is already fetched, we can just reuse it, at least partially (icons might be missing)

**Review by Hyperkid123 — APPROVED** [review]

File: src/js/App/Header/AppFilter.js [hunk 3]
```diff
@@ -153,19 +92,18 @@ const AppFilter = () => {
         />
         <Grid hasGutter>
           <GridItem span={4}>
-            {renderApp(appList[0])}
-            {renderApp(appList[1])}
+            {renderApp(apps?.[0])}
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/js/App/Header/AppFilter.js:95]: @karelhala I would not sweat this ATM. As I see it, the whole concept will probably change 10x. I think we should get it into beta ASAP so the managers can play with it.

File: src/js/App/Header/AppFilter.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import migrationsNamespace from '../../../../static/images/platform-icons/migrations-namespace.svg';
+
+import { getNavFromConfig } from '../../nav/globalNav';
+import sourceOfTruth from '../../nav/sourceOfTruth';
+
+import './AppFilter.scss';
+
+const getIcon = (id = 'default') =>
+  ({
+    insights: <img className="ins-c-app-filter-app-icon" src="/apps/landing/fonts/Insights.svg" alt="Insights Logo" />,
```
Comment (karelhala) [inline src/js/App/Header/AppFilter.js:25]: I would add `isBeta` check in here.

```suggestion
    insights: <img className="ins-c-app-filter-app-icon" src={`${
      insights.chrome.isBeta() ?
      'beta' :
      ''
    }/apps/landing/fonts/Insights.svg" alt="Insights Logo`} />,
```

File: src/js/App/Header/AppFilter.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    openshift: <img className="ins-c-app-filter-app-icon" src={openshift} alt="Openshift Logo" />,
+    SAP: <img className="ins-c-app-filter-app-icon" src="/apps/landing/fonts/SAP.svg" alt="SAP Logo" />,
```
Comment (karelhala) [inline src/js/App/Header/AppFilter.js:27]: Same here as above.

**Review by karelhala — APPROVED** [review]

**Pull request merged**